pkg/admission/decode.go

```diff
@@ -53,7 +57,18 @@ func decodeInfrastructureConfig(decoder runtime.Decoder, infra *core.ProviderCon
 	return infraConfig, nil
 }
 
-func decodeCloudProfileConfig(decoder runtime.Decoder, config *gardencorev1beta1.ProviderConfig) (*gcp.CloudProfileConfig, error) {
+// DecodeCloudProfileConfigProviderConfig decodes the `CloudProfileConfig` from the given `ProviderConfig`.
+func DecodeCloudProfileConfigProviderConfig(decoder runtime.Decoder, config *core.ProviderConfig) (*gcp.CloudProfileConfig, error) {
```

Comment:
`DecodeCloudProfileConfigProviderConfig` sounds a bit duplicated. WDYT about `DecodeCloudProfileConfig` to keep it consistent with the other funcs?
```suggestion
func DecodeCloudProfileConfig(decoder runtime.Decoder, config *core.ProviderConfig) (*gcp.CloudProfileConfig, error) {
```

Reply:
Sorry, I must have missed that.